diff --git a/src/click/types.py b/src/click/types.py
index 0eb8462..22f76f5 100644
--- a/src/click/types.py
+++ b/src/click/types.py
@@ -751,7 +751,7 @@ class IntRange(_NumberRangeBase[int, int], IntParamType):
         if not open:
             return bound
 
-        return bound + dir
+        return bound - dir
 
 
 class FloatParamType(_NumberParamTypeBase[t.SupportsFloat | t.SupportsIndex, float]):
